File: 1.0/src/cz/mg/vulkan/PFNvkCmdBindDescriptorSets.java
package cz.mg.vulkan;

public class PFNvkCmdBindDescriptorSets extends VkFunctionPointer {
    public PFNvkCmdBindDescriptorSets() {
    }

    protected PFNvkCmdBindDescriptorSets(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdBindDescriptorSets(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdBindDescriptorSets(long value) {
        setValue(value);
    }

    public PFNvkCmdBindDescriptorSets(VkInstance instance) {
        super(instance, new VkString("vkCmdBindDescriptorSets"));
    }

    public void call(VkCommandBuffer commandBuffer, int pipelineBindPoint, VkPipelineLayout layout, int firstSet, int descriptorSetCount, VkDescriptorSet pDescriptorSets, int dynamicOffsetCount, VkUInt32 pDynamicOffsets){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pipelineBindPoint, layout != null ? layout.getVkAddress() : VkPointer.getNullAddressNative(), firstSet, descriptorSetCount, pDescriptorSets != null ? pDescriptorSets.getVkAddress() : VkPointer.NULL, dynamicOffsetCount, pDynamicOffsets != null ? pDynamicOffsets.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, int pipelineBindPoint, long layout, int firstSet, int descriptorSetCount, long pDescriptorSets, int dynamicOffsetCount, long pDynamicOffsets);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkCmdBindDescriptorSets.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdBindDescriptorSets_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint pipelineBindPoint, jlong layout, jint firstSet, jint descriptorSetCount, jlong pDescriptorSets, jint dynamicOffsetCount, jlong pDynamicOffsets){
    (void)env;
    (void)jc;
    PFN_vkCmdBindDescriptorSets f = (PFN_vkCmdBindDescriptorSets)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (VkPipelineBindPoint)pipelineBindPoint,
        *((VkPipelineLayout*)jniLongToPointer(layout)),
        (uint32_t)firstSet,
        (uint32_t)descriptorSetCount,
        ((VkDescriptorSet*)jniLongToPointer(pDescriptorSets)),
        (uint32_t)dynamicOffsetCount,
        ((uint32_t*)jniLongToPointer(pDynamicOffsets))
    );
}

